### ISSUE TITLE
lightline.vim cause automatically loadview invalid

### ISSUE THREAD

**Issue by pilgrimlyieu** [itchyny/lightline.vim#636]
I use augroup `auto_view` to memorize the last fold status and cursor position.

```vimrc
augroup auto_view
    autocmd!
    autocmd BufWinLeave *.*    mkview
    autocmd BufWinEnter *.*    silent loadview
    autocmd BufWinLeave _vimrc mkview
    autocmd BufWinEnter _vimrc silent loadview
augroup end
```

However, if I enable `lightline.vim`, it will not restore the last cursor position(fold status is normally restored). Instead, it will focus on the first line.

Here is the minimal vimrc I've tested in [Vim 9.0.0073](https://github.com/vim/vim/commit/1b67f07f7626b87d9ce3e16815970988983a2ddc).(I use `vim-plug` as my vim plugin manager)

```vimrc
call plug#begin("~/vimfiles/plugged")
Plug 'itchyny/lightline.vim'
call plug#end()

augroup auto_view
    autocmd!
    autocmd BufWinLeave *.*    mkview
    autocmd BufWinEnter *.*    silent loadview
    autocmd BufWinLeave _vimrc mkview
    autocmd BufWinEnter _vimrc silent loadview
augroup end
```

Besides, if I disable `lightline.vim` but keep sub-plugin of `lightline.vim`, `auto_view` doesn't work either.

Additionally, if I use `:loadview` command manually, it can work.

If I disable `lightline.vim`, it works normally. Since `lightline.vim` is a wonderful plugin, I wonder that if there is a way to fix it.

By the way, really appreciating your pretty work.

**Comment by itchyny** [comment]
I tested the minimal vimrc but it looks working as intended. What is _sub-plugin_?

**Comment by pilgrimlyieu** [comment]
> What is _sub-plugin_?

sub-plugin is something like `mengelbrecht/lightline-bufferline`. But it is irrelevant.

I've recorded a GIF.

![2](https://user-images.githubusercontent.com/80107081/180981340-ce28157c-bdd5-4814-a80f-df332956d5fa.gif)

**Comment by itchyny** [comment]
I don't still have clue yet but this could be a nested event problem. How about adding `++nested` flag?
```vim
    autocmd BufWinLeave *.*    ++nested mkview
    autocmd BufWinEnter *.*    ++nested silent loadview
```

**Comment by pilgrimlyieu** [comment]
> I don't still have clue yet but this could be a nested event problem. How about adding `++nested` flag?
> 
> ```viml
>     autocmd BufWinLeave *.*    ++nested mkview
>     autocmd BufWinEnter *.*    ++nested silent loadview
> ```

Doesn't work either. :cry:

**Comment by itchyny** [comment]
Removing `silent` shows any errors?

**Comment by pilgrimlyieu** [comment]
> Removing `silent` shows any errors?

Nothing. And if I use `echo` to check, it can print message normally.

```vimrc
augroup auto_view
    autocmd!
    autocmd BufWinLeave *.*    ++nested echo 'success' | sleep 1 | mkview
    autocmd BufWinEnter *.*    ++nested echo 'success' | sleep 1 | loadview
    autocmd BufWinLeave _vimrc ++nested echo 'success' | sleep 1 | mkview
    autocmd BufWinEnter _vimrc ++nested echo 'success' | sleep 1 | loadview
augroup end
```

**Comment by pilgrimlyieu** [comment]
Can you reproduce this issue?

**Comment by itchyny** [comment]
I cannot reproduce the issue.
![](https://user-images.githubusercontent.com/375258/180989207-dc9c99d9-c2ab-43cb-becf-10f06adc19e6.gif)
```vim
set runtimepath+=/tmp/lightline.vim
set laststatus=2

autocmd BufWinLeave * mkview
autocmd BufWinEnter * loadview
```

**Comment by pilgrimlyieu** [comment]
Maybe it is the bug of the plugin manager `vim-plug`?

I change the minimal vimrc to this and it works now.

```vimrc
set runtimepath+=~/vimfiles/plugged/lightline.vim/autoload/lightline.vim
set laststatus=2

autocmd BufWinLeave * mkview
autocmd BufWinEnter * loadview
```

To my vimrc, I should add `set runtimepath+=~/vimfiles/plugged/lightline.vim/autoload/lightline.vim` and disable `lightline.vim` and its sub-plugin under the control of `vim-plug`.

If I don't disable `lightline.vim` or any sub-plugin, it will not work.

However, I could not use some special settings any more.

![Snipaste_2022-07-26_19-22-53](https://user-images.githubusercontent.com/80107081/180994703-645b3e2c-0705-4868-a0d1-5cd8ab3c359f.png)

```vimrc
let s:palette = g:lightline#colorscheme#gruvbox#palette
let s:palette.tabline.ale = [['#282C34', '#92A4A4', 0, 21]]
```

**Comment by pilgrimlyieu** [comment]
I've install [Vim 8.2.4005](https://github.com/vim/vim-win32-installer/releases/tag/v8.2.4005) to check, and it can work fine. So it is the issue of Vim9?

**Comment by itchyny** [comment]
Add the plugin directory, not the autoload file.
```vim
set runtimepath+=~/vimfiles/plugged/lightline.vim
```

**Comment by pilgrimlyieu** [comment]
> 

This take no effect while `set runtimepath+=~/vimfiles/plugged/lightline.vim/autoload/lightline.vim` can.

**Comment by itchyny** [comment]
Please find the exact version Vim which causes the regression.

**Comment by pilgrimlyieu** [comment]
> Please find the exact version Vim which causes the regression.

Through test, I found that [Vim 9.0.0057](https://github.com/vim/vim-win32-installer/releases/tag/v9.0.0057) is the last version that works. And [Vim 9.0.0063](https://github.com/vim/vim-win32-installer/releases/tag/v9.0.0063) is that first version that cause this issue.

So the issue lies in one or some of them.

[9.0.0063](https://github.com/vim/vim/commit/d61efa50f8f5b9d9dcbc136705cc33874f0fdcb3): too many type casts for dict_get functions
[9.0.0062](https://github.com/vim/vim/commit/5ac50de83f1b4136f903c51a1d4e7d84a26c2271): compiler warnings for signed/unsigned char
[9.0.0061](https://github.com/vim/vim/commit/5fa9f23a63651a8abdb074b4fc2ec9b1adc6b089): ml_get error with nested autocommand
[9.0.0060](https://github.com/vim/vim/commit/b9e717367c395490149495cf375911b5d9de889e): accessing uninitialized memory when completing long line
[9.0.0059](https://github.com/vim/vim/commit/bb404f5ad5ec909318bc24e5b82e4ed7b87ba8f4): test file has wrong name
[9.0.0058](https://github.com/vim/vim/commit/81a3ff97e2012bdafc3ece796289f2e11e2754f3): Win32: cannot test low level events

**Comment by itchyny** [comment]
Thank you. I was using a bit older version, but now I can reproduce the issue and git-bisected successfully. Here's the minimum configuration with no dependency on plugin. The bad commit is apparently [9.0.0061](https://github.com/vim/vim/commit/5fa9f23a63651a8abdb074b4fc2ec9b1adc6b089).
```vim
autocmd BufWinLeave * mkview
autocmd BufWinEnter * loadview
autocmd SessionLoadPost * echo
```

**Comment by itchyny** [comment]
Oh, the problem was already reported to https://github.com/vim/vim/issues/10780.

**Comment by itchyny** [comment]
The issue was resolved by [9.0.0094](https://github.com/vim/vim/commit/3d6ee8bda0550a01346f5992bbce09c0eb6d7569).

**Comment by pilgrimlyieu** [comment]
> The issue was resolved by [9.0.0094](https://github.com/vim/vim/commit/3d6ee8bda0550a01346f5992bbce09c0eb6d7569).

I'll install and test after the compiled version released.

Thanks for your effort and patience!

**Comment by pilgrimlyieu** [comment]
It Works, Thanks again!
